video/cloud-client/analyze/beta_snippets.py

```diff
@@ -0,0 +1,207 @@
+        face_detection_config=config)
+
+    operation = video_client.annotate_video(
+        path, features=features, video_context=context)
+    print('\nProcessing video for face annotations:')
+
+    result = operation.result(timeout=900)
+    print('\nFinished processing.')
+
+    # There is only one result because a single video was processed.
+    faces = result.annotation_results[0].face_detection_annotations
+    for i, face in enumerate(faces):
+        print('Face {}'.format(i))
+
+        # Each face_detection_annotation has only one segment.
+        segment = face.segments[0]
+        start_time = (segment.segment.start_time_offset.seconds +
+                      segment.segment.start_time_offset.nanos / 1e9)
+        end_time = (segment.segment.end_time_offset.seconds +
+                    segment.segment.end_time_offset.nanos / 1e9)
+        positions = '{}s to {}s'.format(start_time, end_time)
+        print('\tSegment: {}\n'.format(positions))
+
+        # There are typically many frames for each face,
+        # here we print information on only the first frame.
+        frame = face.frames[0]
+
+        time_offset = (frame.time_offset.seconds +
+                       frame.time_offset.nanos / 1e9)
+        box = frame.attributes[0].normalized_bounding_box
+
+        print('First frame time offset: {}s\n'.format(time_offset))
+
+        print('First frame normalized bounding box:')
+        print('\tleft  : {}'.format(box.left))
+        print('\ttop   : {}'.format(box.top))
+        print('\tright : {}'.format(box.right))
+        print('\tbottom: {}'.format(box.bottom))
+        print('\n')
+# [END video_face_bounding_boxes]
+
+
+# [START video_face_emotions]
+def face_emotions(path):
+    """ Analyze faces' emotions over frames. """
+    video_client = videointelligence.VideoIntelligenceServiceClient()
+    features = [videointelligence.enums.Feature.FACE_DETECTION]
+
+    # include_bounding_boxes must be set to True for include_emotions
+    # to work.
+    config = videointelligence.types.FaceConfig(
+        include_bounding_boxes=True,
+        include_emotions=True)
+    context = videointelligence.types.VideoContext(
+        face_detection_config=config)
+
+    operation = video_client.annotate_video(
+        path, features=features, video_context=context)
+    print('\nProcessing video for face annotations:')
+
+    result = operation.result(timeout=600)
+    print('\nFinished processing.')
+
+    # There is only one result because a single video was processed.
+    faces = result.annotation_results[0].face_detection_annotations
+    for i, face in enumerate(faces):
+        print('Face {}'.format(i))
+
+        frame_emotions = []
+        for frame in face.frames:
+            time_offset = (frame.time_offset.seconds +
+                           frame.time_offset.nanos / 1e9)
+            emotions = frame.attributes[0].emotions
+
+            # from videointelligence.enums
+            emotion_labels = (
```

Comment:
Is there any way to use the videointelligence.enums object directly or pull this programmatically? How badly will this break if the enum changes in the future?

Reply:
The simplest way is for the enums in videointelligence.enums to extend python's Enums class, which would allow us to map the enums back to their names.  However the client libraries do not do that (yet).

I don't know of a clean way to pull the names programmatically, and as it stands now the sample code will break in these possible ways:

1. the order of the enums changes (unlikely), in which case the printout of the sample will be incorrect.

2. more enums are added (possible) and the data we use for testing receive those added enums in the API responses, in which case we would get an IndexError in our testing.

3. (worst) more enums are added, but the testing data does not receive them, so we don't notice it is broken.  Users relying on the same tuple to map enums to names might get IndexError with their data.

The best long term solution is to simply extend Enums.  When that happens we will need to come back and update these samples.

Reply:
+1 Programmatically, this keeps showing up in Python samples and, as a user, it seems really frustrating that Python makes you do this... you get the proto's **enum value index** for enums via the library but not the **enum value name**?

I don't think any of the other languages have this problem?

(Make this better! 😭)

Reply:
Without fixing the client library generation tool, this is the closest I could come up with:
```
from google.cloud import videointelligence_v1p1beta1 as videointelligence
emotion_label = {value:name for name, value in vars(videointelligence.enums.Emotion).iteritems() if not name.startswith('__')}
```

I would rather not have this as is in a code snippet - but perhaps abstracted away as a helper function?  The reader of the sample would not be able to see the full list of enums this way.

Reply:
To be clear, I wasn't thinking about fixing this in the code sample, but rather in google-cloud-core

IMO The code sample should use an inline, literal List with the values. The display name of the enum value should be printed by getting it by index. As we do today.

Reply:
I'll move this to a thread on [google-cloud-core](https://github.com/GoogleCloudPlatform/google-cloud-python/issues) 😄

Reply:
Code in this PR regarding Enum LGTM